Sources/DistributedActors/Time.swift

```diff
@@ -209,187 +165,45 @@ extension TimeAmount: CustomStringConvertible, CustomPrettyStringConvertible {
-}
-
-extension TimeAmount {
-    public static func + (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
-        .nanoseconds(lhs.nanoseconds + rhs.nanoseconds)
-    }
-
-    public static func - (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
-        .nanoseconds(lhs.nanoseconds - rhs.nanoseconds)
-    }
-
-    public static func * (lhs: TimeAmount, rhs: Int) -> TimeAmount {
-        TimeAmount(lhs.nanoseconds * Value(rhs))
-    }
-
-    public static func * (lhs: TimeAmount, rhs: Double) -> TimeAmount {
-        TimeAmount(Int64(Double(lhs.nanoseconds) * rhs))
-    }
-
-    public static func / (lhs: TimeAmount, rhs: Int) -> TimeAmount {
-        TimeAmount(lhs.nanoseconds / Value(rhs))
-    }
-
-    public static func / (lhs: TimeAmount, rhs: Double) -> TimeAmount {
-        TimeAmount(Int64(Double(lhs.nanoseconds) / rhs))
+extension Duration {
+    var toNIO: NIO.TimeAmount {
+        .nanoseconds(self.nanoseconds)
     }
 }
 
 // ==== ----------------------------------------------------------------------------------------------------------------
 // MARK: Deadline
 
-/// Represents a point in time.
-///
-/// - Warning: To be replaced by "proper" types in Swift standard library, currently under debate in [[Pitch] Clock, Instant, Date, and Duration](https://forums.swift.org/t/pitch-clock-instant-date-and-duration/52451/256)
-///
-/// Stores the time in nanoseconds as returned by `DispatchTime.now().uptimeNanoseconds`
-///
-/// `Deadline` allow chaining multiple tasks with the same deadline without needing to
-/// compute new timeouts for each step
-///
-/// ```
-/// func doSomething(deadline: Deadline) -> EventLoopFuture<Void> {
-///     return step1(deadline: deadline).then {
-///         step2(deadline: deadline)
-///     }
-/// }
-/// doSomething(deadline: .now() + .seconds(5))
-/// ```
-///
-/// - note: `Deadline` should not be used to represent a time interval
-public struct Deadline: Equatable, Hashable {
-    // TODO(swift): To be replaced by types that upcoming Swift versions might gain, as discussed in
-    //              [Pitch] Clock, Instant, Date, and Duration https://forums.swift.org/t/pitch-clock-instant-date-and-duration/52451
-
-    public typealias Value = Int64
-
-    /// The nanoseconds since boot representation of the `Deadline`.
-    public let uptimeNanoseconds: Value
+public typealias Deadline = ContinuousClock.Instant
 
-    public init(_ uptimeNanoseconds: Value) {
-        self.uptimeNanoseconds = uptimeNanoseconds
+extension ContinuousClock.Instant {
+    public static var distantPast: ContinuousClock.Instant {
+        .now() - Duration.effectivelyInfinite
     }
 
-    public static let distantPast = Deadline(0)
-    public static let distantFuture = Deadline(Value.max)
-
-    public static func now() -> Deadline {
-        uptimeNanoseconds(Deadline.Value(DispatchTime.now().uptimeNanoseconds))
+    public static var distantFuture: ContinuousClock.Instant {
+        .now() + Duration.effectivelyInfinite
```

Comment:
are these correct?

Reply:
Hmm I think so (and there are tests)? It used to be `Deadline(Int64.max)`.

Reply:
Strike all that I tried it out and it works out since we cap inside the `.nanoseconds`

Thank you!